### ISSUE TITLE
Test compilation fails when using non-default source dirs

### ISSUE THREAD

**Issue by camechis** [gradle-clojure/gradle-clojure#38]
Just started to play around with the new plugin in our project and received this issue when running 

`/gradlew build`  This could be my fault with ignorance around the plugin but just wanted to report it just in case 

```
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':compileTestClojure'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:100)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:70)
        at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:62)
        at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:97)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:87)
        at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.run(DefaultTaskGraphExecuter.java:248)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:241)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:230)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.processTask(DefaultTaskPlanExecutor.java:123)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.access$200(DefaultTaskPlanExecutor.java:79)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:104)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:98)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.execute(DefaultTaskExecutionPlan.java:625)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.executeWithTask(DefaultTaskExecutionPlan.java:580)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:98)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
Caused by: org.gradle.workers.internal.DefaultWorkerExecutor$WorkExecutionException: A failure occurred while executing gradle_clojure.plugin.internal.ClojureWorker
        at org.gradle.workers.internal.DefaultWorkerExecutor$2.waitForCompletion(DefaultWorkerExecutor.java:143)
        at org.gradle.internal.work.DefaultAsyncWorkTracker.waitForItemsAndGatherFailures(DefaultAsyncWorkTracker.java:99)
        at org.gradle.internal.work.DefaultAsyncWorkTracker.access$000(DefaultAsyncWorkTracker.java:34)
        at org.gradle.internal.work.DefaultAsyncWorkTracker$2.run(DefaultAsyncWorkTracker.java:83)
        at org.gradle.internal.work.DefaultWorkerLeaseService.withoutLocks(DefaultWorkerLeaseService.java:196)
        at org.gradle.internal.work.DefaultWorkerLeaseService.withoutProjectLock(DefaultWorkerLeaseService.java:145)
        at org.gradle.internal.work.DefaultAsyncWorkTracker.waitForCompletion(DefaultAsyncWorkTracker.java:79)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:129)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:110)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:92)
        ... 28 more
Caused by: java.lang.ArrayIndexOutOfBoundsException: 0
        at clojure.asm.Type.getObjectType(Type.java:229)
        at clojure.core$generate_class.invokeStatic(genclass.clj:141)
        at clojure.core$gen_class.invokeStatic(genclass.clj:635)
        at clojure.core$gen_class.doInvoke(genclass.clj:507)
        at clojure.lang.RestFn.applyTo(RestFn.java:142)
        at clojure.lang.Var.applyTo(Var.java:702)
        at clojure.lang.Compiler.macroexpand1(Compiler.java:6855)
        at clojure.lang.Compiler.macroexpand(Compiler.java:6915)
        at clojure.lang.Compiler.compile1(Compiler.java:7537)
        at clojure.lang.Compiler.compile1(Compiler.java:7542)
        at clojure.lang.Compiler.compile(Compiler.java:7619)
        at clojure.lang.RT.compile(RT.java:408)
        at clojure.lang.RT.load(RT.java:453)
        at clojure.lang.RT.load(RT.java:421)
        at clojure.core$load$fn__6368.invoke(core.clj:6008)
        at clojure.core$load.invokeStatic(core.clj:6007)
        at clojure.core$load.doInvoke(core.clj:5991)
        at clojure.lang.RestFn.invoke(RestFn.java:408)
        at clojure.core$load_one.invokeStatic(core.clj:5812)
        at clojure.core$compile$fn__6373.invoke(core.clj:6018)
        at clojure.core$compile.invokeStatic(core.clj:6018)
        at clojure.core$compile.invoke(core.clj:6010)
        at gradle_clojure.tools.clojure_compiler$compiler$fn__151.invoke(clojure_compiler.clj:32)
        at gradle_clojure.tools.clojure_compiler$compiler.invokeStatic(clojure_compiler.clj:25)
        at gradle_clojure.tools.clojure_compiler$compiler.invoke(clojure_compiler.clj:23)
        at clojure.lang.AFn.applyToHelper(AFn.java:165)
        at clojure.lang.AFn.applyTo(AFn.java:144)
        at clojure.lang.Var.applyTo(Var.java:702)
        at clojure.core$apply.invokeStatic(core.clj:657)
        at clojure.core$apply.invoke(core.clj:652)
        at clojure.lang.Var.invoke(Var.java:385)
        at org.projectodd.shimdandy.impl.ClojureRuntimeShimImpl.invoke(ClojureRuntimeShimImpl.java:124)
        at org.projectodd.shimdandy.impl.ClojureRuntimeShimImpl.invoke(ClojureRuntimeShimImpl.java:117)
        at gradle_clojure.plugin.internal.ClojureWorker.run(ClojureWorker.java:60)
        at org.gradle.workers.internal.DefaultWorkerServer.execute(DefaultWorkerServer.java:36)
        at org.gradle.workers.internal.WorkerDaemonServer.execute(WorkerDaemonServer.java:46)
        at org.gradle.workers.internal.WorkerDaemonServer.execute(WorkerDaemonServer.java:30)
        at org.gradle.process.internal.worker.request.WorkerAction.run(WorkerAction.java:100)
        at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
        at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
        at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:146)
        at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:128)
        at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:404)
        ... 3 more
```

Also note since this is a lein project as well I have set these sourceSets

```groovy
sourceSets {
    dev {
        clojure.srcDirs = ['dev']
        resources.srcDirs = ['dev-resources']
    }
    main {
        clojure.srcDirs = ['src']
        resources.srcDirs = ['resources']
    }
    test {
        clojure.srcDirs = ['test']
    }
}
```

**Comment by ajoberstar** [comment]
Thanks for trying the plugin out. Nothing's popping out to me from that stack trace. Was there any other output in the build that might indicate which file was being compiled at the time? FYI, we generate classes for JUnit runners as part of test compile, so it very well could be a bug in that code as opposed to failing to compile one of your test namespaces.

If possible, could you provide the project (or another sample that reproduces this)?

**Comment by camechis** [comment]
I don't see anything relating to our code specifically.  Unfortunately I cannot share this project.  If you have any thoughts I would be happy share any snippets or try anything.

**Comment by neuromantik33** [comment]
I'm getting the same problem when compiling tests. If you want I can create a test project to reproduce the issue. FYI I'm also using Clojure 1.9.0-beta4.

**Comment by camechis** [comment]
That would be great I think if you have the time

**Comment by camechis** [comment]
I did some experimenting and I think it might have something to do with the classpath/sourcesets.  I moved all my tests out of my project and made sure the test task could run with no tests.  I then moved the simplest of tests I have ( very basic ) and it failed again.  I then took the simple test from the sample project and it also failed with in my project.  My next step was to move it into the src/test/clojure directory and remove my source sets.  The test then ran just fine without the exception.
